### PR TITLE
[Numpy]add row_stack (= vstack)

### PR DESCRIPTION
**Pull request by hanke580**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/symbol/numpy/_symbol.py [hunk 1]
```diff
@@ -42,7 +42,7 @@
            'rint', 'radians', 'reciprocal', 'square', 'negative', 'fix', 'ceil', 'floor',
            'trunc', 'logical_not', 'arcsinh', 'arccosh', 'arctanh', 'tensordot', 'histogram', 'eye',
            'linspace', 'logspace', 'expand_dims', 'tile', 'arange', 'split', 'vsplit', 'concatenate', 'append',
-           'stack', 'vstack', 'column_stack', 'dstack', 'mean', 'maximum', 'minimum', 'swapaxes', 'clip', 'argmax',
+           'stack', 'vstack','row_stack', 'column_stack', 'dstack', 'mean', 'maximum', 'minimum', 'swapaxes', 'clip', 'argmax',
```
Comment (xidulu) [inline python/mxnet/symbol/numpy/_symbol.py]: 'vstack',{1 empty space}'row_stack',

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -3131,6 +3131,55 @@ def get_list(arrays):
...
+            [3.],
+            [4.]])
+    """
+    def get_list(arrays):
+        if not hasattr(arrays, '__getitem__') and hasattr(arrays, '__iter__'):
+            raise ValueError("expected iterable for arrays but got {}".format(type(arrays)))
+        return [arr for arr in arrays]
+
+    arrays = get_list(arrays)
+    return _npi.vstack(*arrays)
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: one more blank line after this line.

File: python/mxnet/numpy/multiarray.py [hunk 3]
```diff
@@ -5030,6 +5030,49 @@ def vstack(arrays, out=None):
...
+    >>> b = np.array([[2], [3], [4]])
+    >>> np.vstack((a, b))
+    array([[1.],
+           [2.],
+           [3.],
+           [2.],
+           [3.],
+           [4.]])
+    """
+    return _mx_nd_np.row_stack(arrays)
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: same here.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 4]
```diff
@@ -3244,6 +3244,39 @@ def get_list(arrays):
...
+    """
+    def get_list(arrays):
+        if not hasattr(arrays, '__getitem__') and hasattr(arrays, '__iter__'):
+            raise ValueError("expected iterable for arrays but got {}".format(type(arrays)))
+        return [arr for arr in arrays]
+
+    arrays = get_list(arrays)
+    return _npi.vstack(*arrays)
+
+
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: 1 less blank line here.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 5]
```diff
@@ -3244,6 +3244,39 @@ def get_list(arrays):
     arrays = get_list(arrays)
     return _npi.vstack(*arrays)
 
+@set_module('mxnet.symbol.numpy')
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: one more blank line above.

**Comment by haojin2** [comment]
@hanke580 Please rebase with master and resolve the conflicts and code review comments.

File: python/mxnet/numpy/multiarray.py (hunk 3 above)
Comment (hanke580) [reply python/mxnet/numpy/multiarray.py]: Fixed, thanks

File: python/mxnet/ndarray/numpy/_op.py (hunk 2 above)
Comment (hanke580) [reply python/mxnet/ndarray/numpy/_op.py]: Fixed, thanks

File: python/mxnet/symbol/numpy/_symbol.py (hunk 1 above)
Comment (hanke580) [reply python/mxnet/symbol/numpy/_symbol.py]: Fixed, thx

**Pull request closed without merging**